precommit/src/main/shell/core.d/01-common.sh

```diff
@@ -303,6 +303,27 @@ function common_args
   USER_PLUGIN_DIR="${BASEDIR}/.yetus/plugins.d"
 }
 
+## @description  Verify that BASEDIR is a git repo
+## @description  and set some git settings
+## @audience     public
+## @stability    evolving
+## @replaceable  no
+function verify_basedir_repo
+{
+  if [[ ! -e "${BASEDIR}/.git" ]]; then
+    yetus_error "ERROR: ${BASEDIR} is not a git repo."
+    cleanup_and_exit 1
```

Comment:
Forgive my ignorance. Is it a common pattern in our public API that functions prefixed with `verify_` can exit the process when their condition is not met?

Reply:
Great question. This one might be the only one.  I should probably rename it from verify to ... something else.. since it isn't really a verify function.

Reply:
Renamed to `check_` which makes a lot more sense.